pkg/controllers/flannelmigration/ipam_migrator.go

```diff
@@ -252,8 +252,10 @@ func setupCalicoNodeVxlan(ctx context.Context, c client.Interface, nodeName stri
 
 	log.Infof("Calico Node current value: %+v.", node)
 
-	node.Spec.BGP = &api.NodeBGPSpec{} // set public ip for node
-	node.Spec.BGP.IPv4Address = publicIP
+	node.Spec.BGP = &api.NodeBGPSpec{}
+	// Set public ip with subnet /32.
+	// The subnet part is required to pass Felix validation.
+	node.Spec.BGP.IPv4Address = fmt.Sprintf("%s/32", publicIP)
```

Comment:
Will flannel ever write subnet information here? i.e., do we need to handle if the `publicIP` already has a subnet?

Reply:
I am pretty confident that Flannel just write ip without subnet information. https://github.com/coreos/flannel/blob/master/subnet/subnet.go#L37 This part of code has not been updated since 2017.